driver-dse/src/main/java/com/datastax/driver/auth/DseAuthProvider.java

```diff
@@ -65,20 +91,153 @@
  *       principal="user@MYDOMAIN.COM";
  * };
  * </pre>
- * <p/>
- * To connect to clusters using internal authentication, use the standard method for
- * setting credentials. eg:
- * <p/>
- * <pre>
- * Cluster cluster = Cluster.builder()
- *                          .addContactPoint(hostname)
- *                          .withCredentials("username", "password")
- *                          .build();
- * </pre>
  */
 public class DseAuthProvider implements AuthProvider {
-    @Override
+    private static final String DSE_AUTHENTICATOR = "com.datastax.bdp.cassandra.auth.DseAuthenticator";
+    private final String username;
+    private final String password;
+
+    public DseAuthProvider() {
```

Comment:
It might be useful to users to provide javadocs for both constructors, specially because using one or the other induces the authentication mechanism used behind the scenes.
Or alternatively, one could make this class abstract and provide to small subclasses, `PlainTextDseAuthProvider` and `GSSAPIDseAuthProvider`, to make things even clearer for users.